src/cmds/qsub.c

```diff
@@ -1833,7 +1833,7 @@ read_job_script(char *script)
 	int errflg; /* error code from get_script() */
 	struct stat statbuf;
 	char *bnp;
-	char basename[PBS_MAXJOBNAME + 1]; /* base name of script for job name*/
+	char basename[MAXPATHLEN + 1]; /* base name of script for job name*/
```

Comment:
why did you change this? PBS_MAXJOBNAME is supposed to be used for this exact purpose.

Reply:
basename holds script file that is static char script[MAXPATHLEN + 1] = "";

Reply:
then we should change PBS_MAXJOBNAME to be MAXPATHLEN + \<N>.

Reply:
@agrawalravi90 PBS_MAXJOBNAME is not full path of job script, its max job name (`-N` in qsub/pbs_rsub) which is derived from job script's path if `-N` not given. So I don't agree with you that we should change PBS_MAXJOBNAME.

Reply:
If job name is derived from job script's path then does it not make sense that the job name buffer be at least MAXPATHLEN in size? Anyways, the variable name and comment here seemed to suggest that it should use buffer size for job name, but I'm ok with anything as long as compiler clears it.